Source/Scene/Primitive.js

```diff
@@ -283,6 +285,28 @@ define([
         }
         //>>includeEnd('debug');
 
+        /**
+         * Determines whether this primitive will cast shadows when shadow mapping is enabled.
+         *
+         * @type {Boolean}
+         *
+         * @default true
+         */
+        this.castShadows = defaultValue(options.castShadows, true);
+        this._castShadows = this.castShadows;
+
+        /**
+         * Determines whether this primitive will receive shadows when shadow mapping is enabled.
+         *
+         * @type {Boolean}
+         *
+         * @default true
+         */
+        this.receiveShadows = defaultValue(options.receiveShadows, true);
+        this._receiveShadows = this.receiveShadows;
+
+        this._shadowsEnabled = true;
```

Comment:
Minor, and up to you, but I didn't know what this was when I first saw it.  Maybe `_frameStateShadowsEnable` or `_sceneShadowsEnabled` is a better name in this context of this file.